### PR TITLE
chore: add 'optionalDependencies' to package.json

### PR DESCRIPTION
**Pull request by heitorado**
## What's the purpose of this pull request?
This PR adds the `optionalDependencies` section at `package.json` and adds dependencies that are not necessary to build the application to it.
This was made so the build time is reduced, since we are now skipping unnecessary build dependencies such as `eslint` and `storybook`. A faster yarn install can now be performed with `yarn install --ignore-optional`

Benchmarks (at my local development machine):
| Command                                                 | Output            | Note                                                                                                    |
|---------------------------------------------------------|-------------------|---------------------------------------------------------------------------------------------------------|
| `rm -rf node_modules && yarn install`                   | ✨ Done in 82.41s. | This was the command we were running before when building the application. All packages are installed.  |
| `rm -rf node_modules && yarn install --ignore-optional` | ✨ Done in 48.51s. | This is the new command that we intend to use in future webops builds, ignoring packages in `optionalDependencies` |

We can observe a 41.13% reduction in the execution of `yarn install` when using the `--ignore-optional` flag

## How does it work?
By using the `optionalDependencies` category, we can now only install packages that we are really going to use during build time.

## How to test it?
Basically, replicate the benchmarks mentioned in the first section. Here's a guide:
- Clear node_modules
- Run `yarn install`, with and without the `--ignore-optional` flag
- Verify that the package install time decreases when using the flag
- Run `yarn build`
- Verify that the build completes successfully without the optional packages

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,50 +56,52 @@
   },
```
Comment (filipewl) [inline package.json:56]: I don't know much about this, but when WebOps builds for production, doesn't it already [skip](https://classic.yarnpkg.com/en/docs/cli/install/#toc-yarn-install-production-true-false) all the `devDependencies` because `NODE_ENV` is set to `production` (I'd hope)?

Comment (gvc) [reply package.json:56]: @filipewl It does not because it has to run the tests and the lighthouse checks.

Comment (filipewl) [reply package.json:56]: But for building the static pages, I don't think they need to install the dev dependencies...

The tests and LH checks are run on their own separate pipeline run and in these cases they do install the dev dependencies to run the proper commands.

Comment (heitorado) [reply package.json:56]: @filipewl, indeed the devDependencies are not needed for building the static pages, as well as the tests and LH checks have separate pipelines.

However, if you look at our tekton dashboard and browse the pipelines' code, you'll notice that they share the same common task, called `clone`. This `clone` task is responsible for, among other things, running `yarn install` for the cloned repo in the CI environment. Since this is shared between pipelines, we need to fill the `optionalDependencies` with the packages used in all pipelines, letting out only things we use locally, such as the `storybook` and `eslint`.

I agree that different tasks like `clone-for-test` and `clone-for-build` would be better, but:
- `yarn` does not support other sections besides `optionalDependencies`. If they had support for `testDependencies` and `buildDependencies` that would be great.
- The impact on the time is minimal as per the benchmarks I've run (adding cypress and lighthouse dependencies adds ~4s comparing to when only using the build dependencies)

We could investigate alternatives about to reduce the install time even further, maybe at webops level we could run different `yarn install` commands depending on the pipeline. But that is a story for another PR :)

**Comment by heitorado** [comment]
Converted this to draft until [webops#84](https://github.com/vtex/webops-infra/pull/84) is merged and I can test the full build pipeline to evaluate which `cypress` and `lighthouse` dependencies need to be included as well.

**Comment by tlgimenes** [comment]
Make sure `@next/swc` keeps working after these changes too
